### DISCUSSION TITLE
请问用Reality回落多个自建站点的正确方式应该是什么？

### DISCUSSION THREAD

**Discussion by Benjamin1919** [XTLS/Xray-core#2246] [Q&A] [answered]
需求：只有一台VPS，但是想建两个站点（根域名相同，已申请根域名证书），暂且将两个站点称为“1.abc.com"和"2.abc.com"。
参考chika大神[“自己偷自己”](https://github.com/chika0801/Xray-examples/tree/main/VLESS-XTLS-uTLS-REALITY/steal_yourself)的模板，我写了一下服务端reality和nginx的配置，请大佬们勘误，谢谢！

### xray入站配置：

    "inbounds": [
        {
            "listen": "0.0.0.0",
            "port": 443,
            "protocol": "vless",
            "settings": {
                "clients": [
                    {
                        "id": "yourid", 
                        "flow": "xtls-rprx-vision"
                    }
                ],
                "decryption": "none"
            },
            "streamSettings": {
                "network": "tcp",
                "security": "reality",
                "realitySettings": {
                    "show": false, 
                    "dest": ["6001", "6002", "6003", "6004"],
                    "xver": 1,
                    "serverNames": [
                        "1.abc.com",
                        "2.abc.com"
                    ],
                    "privateKey": "2KZ4uouMKgI8nR-LDJNP1_MHisCJOmKGj9jUjZLncVU", 
                    "shortIds": [
                        "6ba85179e30d4fc2"
                    ]
                }
            },
            "sniffing": {
                "enabled": true,
                "destOverride": [
                    "http",
                    "tls",
                    "quic"
                ]
            }
        }
    ]

### Nginx部分配置：

    http {
        set_real_ip_from            127.0.0.1;
        real_ip_header              proxy_protocol; 
        #其他配置省略

        server {
            listen                  127.0.0.1:6001 ssl proxy_protocol;
            listen                  127.0.0.1:6002 ssl http2 proxy_protocol;
            server_name             1.abc.com;

            ssl_certificate         /etc/ssl/private/fullchain.cer;
            ssl_certificate_key     /etc/ssl/private/private.key;
            ssl_protocols           TLSv1.2 TLSv1.3;
            ssl_session_timeout     1d;
            ssl_session_cache       shared:SSL:10m;
            ssl_session_tickets     off;
            ssl_stapling            on;
            ssl_stapling_verify     on;
            ssl_trusted_certificate /etc/ssl/private/fullchain.cer;
            #location配置省略
        }
    
        server {
            listen                  127.0.0.1:6003 ssl proxy_protocol;
            listen                  127.0.0.1:6004 ssl http2 proxy_protocol;
            server_name             2.abc.com;

            ssl_certificate         /etc/ssl/private/fullchain.cer;
            ssl_certificate_key     /etc/ssl/private/private.key;
            ssl_protocols           TLSv1.2 TLSv1.3;
            ssl_session_timeout     1d;
            ssl_session_cache       shared:SSL:10m;
            ssl_session_tickets     off;
            ssl_stapling            on;
            ssl_stapling_verify     on;
            ssl_trusted_certificate /etc/ssl/private/fullchain.cer;
            #location配置省略
        }
    }

**Comment by lxhao61** [comment, marked as answer]
配置修改如下：
xray部分配置修改见如下：
`"dest": ["6001", "6002", "6003", "6004"],`改成`"dest": 6000,`
Nginx部分配置修改后见如下：
```
http {
    set_real_ip_from            127.0.0.1;
    real_ip_header              proxy_protocol; 
    #其他配置省略

    server {
        listen                  127.0.0.1:6000 ssl http2 proxy_protocol;
        server_name             1.abc.com;

        ssl_certificate         /etc/ssl/private/fullchain.cer;
        ssl_certificate_key     /etc/ssl/private/private.key;
        ssl_protocols           TLSv1.2 TLSv1.3;
        ssl_session_timeout     1d;
        ssl_session_cache       shared:SSL:10m;
        ssl_session_tickets     off;
        ssl_stapling            on;
        ssl_stapling_verify     on;
        ssl_trusted_certificate /etc/ssl/private/fullchain.cer;
        #location配置省略
    }

    server {
        listen                  127.0.0.1:6000 ssl http2 proxy_protocol; 
        server_name             2.abc.com;

        ssl_certificate         /etc/ssl/private/fullchain.cer;
        ssl_certificate_key     /etc/ssl/private/private.key;
        ssl_protocols           TLSv1.2 TLSv1.3;
        ssl_session_timeout     1d;
        ssl_session_cache       shared:SSL:10m;
        ssl_session_tickets     off;
        ssl_stapling            on;
        ssl_stapling_verify     on;
        ssl_trusted_certificate /etc/ssl/private/fullchain.cer;
        #location配置省略
    }
}
```
注意：上面 Nginx 配置为版本小于 v1.25.1 的一般配置，若版本不小于 v1.25.1 需调整配置；优化及调整参考 [Xray(M+H+K+G+B+A)+Nginx](https://github.com/lxhao61/integrated-examples/tree/main/Xray(M%2BH%2BK%2BG%2BB%2BA)%2BNginx) 示例。

**Reply by Benjamin1919** [reply]
谢谢大佬！再请教一个可能比较小白的问题😂：
nginx配置里只保留“listen  127.0.0.1:6000 ssl http2 proxy_protocol”，是否还能通过http1.1来访问网站？

**Reply by lxsq** [reply]
截止到现在Stable版本H2可以，H2C不行。1.25.1之后都可以。

**Reply by lxhao61** [reply]
@Benjamin1919 
Nginx 加 HTTP 自动跳转 HTTPS 配置。HTTP/2 向下兼容 HTTPS。

**Reply by lxhao61** [reply]
@lxsq
你说的跟 Reality 应用转发给网站（HTTP/2）无关，仅跟回落（解密TLS后转发）到网站（H2C或HTTP/1.1）有关。

**Comment by Amanda-Fisher** [comment]
感谢各位大佬，按照这个配置亲测可行～

**Comment by jliveusa** [comment]
请问使用caddy回落多个站点该如何修改，thx～

**Reply by chika0801** [reply]
我不用Caddy，所以我也不知道

**Reply by askdnl** [reply]
> 我不用Caddy，所以我也不知道

chika大佬你好，我也是用你的模板nginx自己偷自己，也需要根据reality的加分项转发tcp80 和udp443端口吗？
谢谢

**Reply by lxhao61** [reply]
> 请问使用caddy回落多个站点该如何修改，thx～

参考 [Caddy 的特色应用配置方法](https://github.com/lxhao61/integrated-examples/tree/main/Caddy(Other%20Configuration))  中 ``六、Caddy 的转发或回落到不同网站配置方法``。

**Reply by chika0801** [reply]
> > 我不用Caddy，所以我也不知道
> 
> chika大佬你好，我也是用你的模板nginx自己偷自己，也需要根据reality的加分项转发tcp80 和udp443端口吗？ 谢谢

这个“传说”我以前也试过，做太累了，我自己是早没加这2个。看你自己，不想折腾建议不管它，想听官方上写的建议就研究研究，虽然 我也是折腾过的人。

**Reply by askdnl** [reply]
> > > 我不用Caddy，所以我也不知道
> > 
> > chika大佬你好，我也是用你的模板nginx自己偷自己，也需要根据reality的加分项转发tcp80 和udp443端口吗？ 谢谢
> 
> 这个“传说”我以前也试过，做太累了，我自己是早没加这2个。看你自己，不想折腾建议不管它，想听官方上写的建议就研究研究，虽然 我也是折腾过的人。
>感谢大佬回答，我以前是偷别人，然后看您以前关于此问题的回答，用iptables转发到目标的ip，这偷自己我寻思应该不用转发了吧

**Reply by chika0801** [reply]
> ，这偷自己我寻思应该不用转发了吧

你说到偷自己的话，80端口我习惯是做的跳转443。然后UDP的443是我也没特别管它（想过它怎么样，你不问我都忘记了，建议你也别纠结了）

**Reply by askdnl** [reply]
> > ，这偷自己我寻思应该不用转发了吧
> 
> 你说到偷自己的话，80端口我习惯是做的跳转443。然后UDP的443是我也没特别管它（想过它怎么样，你不问我都忘记了，建议你也别纠结了）

ok谢谢大佬解惑。
